osbuild/sources.py

```diff
@@ -50,6 +98,9 @@ def make_items_file(self, tmp):
 class SourceService(host.Service):
     """Source host service"""
 
+    def send_sub_result(self, item: Dict):
```

Comment:
Also in the current form it is not super useful. One way would be to have specialised functions that would encapsulate the actual formatting and signalling, like `.signal_error(checksum, SourceError.TIMEOUT, msg)`

Reply:
What I wanted for the design at the beginning:
* make the sub_command hold the responsibility of defining it's own errors without having to centralize the knowledge on the more abstract souces.py file
* keep the list of result for monitoring purposes.

Since we delayed the monitoring, it makes no sense to unpack a result. I would have prefer a way for downloading commands to define their error set. So we can add more of them without amending sources.py in the future. But what you propose works for me, a list of error kinds written in the sources.py file that a sub_command can use to communicate what went wrong.